### PR TITLE
diff: add integration test

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
sync_diff_inspector lack integration test

### What is changed and how it works?
1. refine the tests structure
2. add a integration test for sync_diff_inspector, now only add test for snapshot and shard, will add more test later

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+
+This folder contains all tests which relies on external service such as TiDB.
+
+## Preprations
```
Comment (csuzhangxc) [inline tests/README.md]:
```suggestion
## Preparations
```

File: tests/_utils/check_db_status [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+for i in {1..20}
+do
+    if mysqladmin -h "$1" -P "$2" -u root --default-character-set utf8 ping > /dev/null 2>&1
```
Comment (csuzhangxc) [inline tests/_utils/check_db_status:9]: check `mysqladmin` existing in _Makefile_ or mention it in _README.md_.

File: tests/_utils/check_db_status [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (csuzhangxc) [inline tests/_utils/check_db_status:1]: add some comments about arguments for these scripts?

File: tests/run.sh [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    echo "Running test $script..."
+    echo "*******************************************"
+    PATH="$pwd/../bin:$pwd/_utils:$PATH" \
+    OUT_DIR=$OUT_DIR \
+    TEST_NAME=$test_name \
+    sh "$script"
+done
+
+# with color
+echo "\033[0;36m<<< Run all test success >>>\033[0m"
```
Comment (csuzhangxc) [inline tests/run.sh:99]: _color_ can work in CI?

File: tests/run.sh [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+rm -rf $OUT_DIR || true
+
+stop_services() {
+    killall -9 tikv-server || true
+    killall -9 pd-server || true
+    killall -9 tidb-server || true
+}
+
+start_upstream_tidb() {
```
Comment (csuzhangxc) [inline tests/run.sh]: not used?

File: tests/sync_diff_inspector/shard/run.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+OUT_DIR=/tmp/tidb_tools_test/sync_diff_inspector
+
+echo "generate data to sharding tables"
+mysql -uroot -h 127.0.0.1 -P 4001 -e "create table diff_test.shard_test1(a int, b varchar(10), c float, d datetime, primary key(a));"
+mysql -uroot -h 127.0.0.1 -P 4001 -e "create table diff_test.shard_test2(a int, b varchar(10), c float, d datetime, primary key(a));"
+
+# each table only have part of data
+mysql -uroot -h 127.0.0.1 -P 4001 -e "insert into diff_test.shard_test1 (a, b, c, d) SELECT a, b, c, d FROM diff_test.test WHERE a%2=0"
+mysql -uroot -h 127.0.0.1 -P 4001 -e "insert into diff_test.shard_test2 (a, b, c, d) SELECT a, b, c, d FROM diff_test.test WHERE a%2=1"
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/shard/run.sh:15]: where do we _load_ these data into target TiDB?

File: tests/_utils/check_db_status (hunk 3 above)
Comment (WangXiangUSTC) [reply tests/_utils/check_db_status:1]: done

File: tests/_utils/check_db_status (hunk 2 above)
Comment (WangXiangUSTC) [reply tests/_utils/check_db_status:9]: done

File: tests/run.sh (hunk 4 above)
Comment (WangXiangUSTC) [reply tests/run.sh:99]: no, but we can run test in our local environment

File: tests/run.sh (hunk 5 above)
Comment (WangXiangUSTC) [reply tests/run.sh]: removed

File: tests/sync_diff_inspector/shard/run.sh (hunk 6 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/shard/run.sh:15]: load in tests/sync_diff_inspector/run.sh

Comment (csuzhangxc) [reply tests/sync_diff_inspector/shard/run.sh:15]: that before this scripts? and where do we _load_ data in `shard_test1` and `shard_test2`?

Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/shard/run.sh:15]: yes, first will generate base table `diff_test.test` in `tests/sync_diff_inspector/run.sh`. And in shard/run.sh, will generate shard tables' data by select `diff_test.test`

Comment (csuzhangxc) [reply tests/sync_diff_inspector/shard/run.sh:15]: Oh, I know.

**Review by csuzhangxc — COMMENTED** [review]
LGTM

File: tests/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+
+
+This folder contains all tests which relies on external service such as TiDB.
+
+## Preparations
+
+1. The following three executables must be copied or linked into these locations:
```
Comment (suzaku) [inline tests/README.md]: There are more than `three` files listed below.

File: tests/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+## Running
+
+Run `make integration_test` to execute the integration tests. This command will
+
+1. Build binaries.
+2. Check that all executables exist.
+3. Execute `tests/run.sh`
+
+If the first two steps are done before, you could also run `tests/run.sh directly.
```
Comment (suzaku) [inline tests/README.md]:
```suggestion
If the first two steps are done before, you could also run `tests/run.sh` directly.
```

File: tests/README.md (hunk 7 above)
Comment (WangXiangUSTC) [reply tests/README.md]: updated

**Review by suzaku — APPROVED** [review]
LGTM

**Pull request merged**